GSA/uswds-sf-lightning-community
# Issue 83

**Summary**



This PR fixes/implements the following **bugs/features**

- [x] #83 
- [x] #53 

Explain the **motivation** for making this change. What existing problem does the pull request solve?

## Test plan (required)

### Installation

- Install parent uswds-sf-lightning-community package `ant deployCode -Dsf.deployRoot /src`
- Install package `ant deployCode -Dsf.deployRoot /examples/uswdsCard`
- Assign `ManagedContentControllerForLex` to the guest user profile of the test community

#### Create CMS Content

- Before creating content, you may need to assign Salesforce CMS tab access to your user profile.
- Edit profile and update each of the following tabs to "Default On"
  - CMS Channels
  - CMS Experiences
  - CMS Home
  - CMS Workspaces
- Turn on app visibility for Salesforce CMS under Custom App Settings
- SAVE
- Navigate to Salesforce CMS App from App Picker
- Create new CMS Workspace (give it any name you like)
- Associate the new Workspace to your existing community.
- Skip adding Contributors
- Select language and default
- DONE
- Create one or more new CMS articles of the News content type. Enter data and images into all fields and publish. No new fields need to be created.
- Navigate to the Community Workspace (CMS Home > All Experiences > Workspace)
- Click "Content Management"
- Click "Collections" tab
- Create new collection specific to the News content type
- How would you like to curate your collection? >> Conditionally
- Filter By: Publish Date, When: Greater Than, Value: 01/01/2021
- SAVE
- Navigate to Experience Builder
- Drag "ex Cards CMS - USWDS" to the page
- Publish
- Click "Get Content" button, articles should appear as cards with click-through to full article.

#### Test Error Handling

  - Update one or more object keys in the `exampleCardWrapperHelper.js` file. For example body > bodyA. 
  - Update cardType value for one or more keys. For example `cardType: 'Card'` >> `cardType: 'CardiB'`
  - Deploy /examples/uswdsCard `ant deployCode -Dsf.deployRoot "/examples/uswdsCard"`

  - [x] Observe error handling via Toast messages. One or more messages should appear for each update made above. 
  - [x] Observe error messages in Developer Console for each Toast message.
  - [x] Revert changes made to `exampleCardWrapperHelper.js` `git checkout examples/uswdsCard/aura/exampleCardWrapper/exampleCardWrapperHelper.js`
  - [x] Redeploy /examples/uswdsCard to undo change -  `ant deployCode -Dsf.deployRoot "/examples/uswdsCard"`

**Code formatting**

✅ 

**Closing issues**

Closes #83
